src/Core/ExecutionPathTracer/ExecutionPathTracer.cs

```diff
@@ -55,15 +57,36 @@ public void OnOperationStartHandler(ICallable operation, IApplyData arguments)
         {
             this.currentDepth++;
 
+            // If `compositeTracer` is initialized, pass operations into it instead
+            if (this.compositeTracer != null)
+            {
+                this.compositeTracer.OnOperationStartHandler(operation, arguments);
+            }
             // Parse operations at or above specified depth
-            if (this.currentDepth <= this.renderDepth)
+            else if (this.currentDepth <= this.renderDepth)
```

Comment:
Similar to the above comment, you could consider something like:
```
if (this.currentDepth > this.renderDepth) return;
```
such that all of the code below can be un-indented. Again, this is stylistic - for readability, I tend to prefer short-circuit return statements rather than long indented if blocks - feel free to ignore.